client/ayon_shotgrid/addon.py

```diff
@@ -80,6 +80,18 @@ def is_local_storage_enabled(self):
     def get_local_storage_key(self):
         return self._local_storage_key or None
 
+    def get_credentials(self):
+        from .lib import credentials
+
+        if self._client_login_type == "env":
```

Comment:
I think we are missing here check, if we are on farm. If we are, we should depend on `AYON_SG_USERNAME` env var as that is filled by Shotgrid AND is passed via `submit_publish_job`.

We need just double check in local publishing that `AYON_SG_USERNAME` contains correct value.

We might use check against `HEADLESS_PUBLISH` env var to differentiate, that we are on farm right now.

Reply:
I added check against `HEADLESS_PUBLISH`, tested, it seems to be working.